### PR TITLE
chore(updatecli) Track JDK version

### PR DESCRIPTION
**Pull request by jayfranco999**
Locally tested and updated jdk17 using updatecli manifest

### CODE REVIEW

**Comment by dduportal** [comment]
Quick message: as discussed with @jayfranco999 , I'm taking over the issue here and continuing his work.

**Comment by dduportal** [comment]
Update: resolved conflicts as below:

- Before:

Capture d’écran 2024-06-17 à 15 44 36


Capture d’écran 2024-06-17 à 16 04 10



- Resolving:

```
$ gh pr checkout 115 #
$ git fetch --all --prune --tags                                                                     
$ git rebase upstream/main 
warning: skipped previously applied commit 9318607
hint: use --reapply-cherry-picks to include skipped commits
hint: Disable this message with "git config advice.skippedCherryPicks false"
Successfully rebased and updated refs/heads/jayfranco999/main.
$ git push git@github.com:jayfranco999/docker-packaging.git HEAD:main --force           
```

File: updatecli/values.yaml [hunk 1]
```diff
@@ -7,3 +7,6 @@ github:
   branch: "main"
   owner: "jenkins-infra"
   repository: "docker-packaging"
+
+jdk:
+  majorVersion: 17
```
Comment (lemeurherve) [inline updatecli/values.yaml:12]: Optional suggestion: WDYT about putting this value in https://github.com/jenkins-infra/docker-packaging/blob/main/Makefile?

**Comment by dduportal** [comment]
Updated:
- Specifying the major line of JDK in the `values.yaml` file (advanced trick ;) )
- Updating the manifest to use the new value above
- Removed the condition checking if there is an `ARG` (too verbose for this new manifest)
- Updated the Docker image published check to automatically retrieved the Docker image tag suffix (and avoid maintaining it)
- Added a target to update the major JDK in the test harness (`cst.yml`)
- Cherry-picked the `Dockefile` changes in https://github.com/jenkins-infra/docker-packaging/pull/116 to have a working check here

**Review by lemeurherve — APPROVED** [review]
Local test OK, LGTM

File: updatecli/values.yaml (hunk 1 above)
Comment (dduportal) [reply updatecli/values.yaml:12]: What would be the goal of specifying it in the `Makefile`? Not sure of the goal (and how it would work)

Comment (lemeurherve) [reply updatecli/values.yaml:12]: Or using

> Added a target to update the major JDK in the test harness (cst.yml)

Maybe not applicable?

Comment (dduportal) [reply updatecli/values.yaml:12]: I don't understand what you are asking?

Comment (lemeurherve) [reply updatecli/values.yaml:12]: Discussed together, this version concerns only updatecli, its current location makes sense.

**Review by dduportal — APPROVED** [review]
```
19:00:46  TARGETS
19:00:46  ========
19:00:46  
19:00:46  updateDockerfileMajorVersion
19:00:46  ----------------------------
19:00:46  
19:00:46  **Dry Run enabled**
19:00:46  
19:00:46  ⚠ The line #4, matched by the keyword "ARG" and the matcher "BUILD_JDK_MAJOR", is changed from "ARG BUILD_JDK_MAJOR=11" to "ARG BUILD_JDK_MAJOR=17".
19:00:46  ⚠ The line #98, matched by the keyword "ARG" and the matcher "BUILD_JDK_MAJOR", is changed from "ARG BUILD_JDK_MAJOR" to "ARG BUILD_JDK_MAJOR=17".
19:00:46  ⚠ - changed lines [4 98] of file "/tmp/updatecli/github/jenkins-infra/docker-packaging/Dockerfile"
19:00:47  
19:00:47  
19:00:47  updateDockerfileVersion
19:00:47  -----------------------
19:00:47  
19:00:47  **Dry Run enabled**
19:00:47  
19:00:47  ⚠ The line #2, matched by the keyword "ARG" and the matcher "JAVA_VERSION", is changed from "ARG JAVA_VERSION=11.0.20_8" to "ARG JAVA_VERSION=17.0.11_9".
19:00:47  ⚠ - changed lines [2] of file "/tmp/updatecli/github/jenkins-infra/docker-packaging/Dockerfile"
19:00:47  
19:00:47  updateTestHarness
19:00:47  -----------------
19:00:47  
19:00:47  **Dry Run enabled**
19:00:47  
19:00:47  ⚠ - change detected:
19:00:47  	* key "$.commandTests[1].expectedOutput[0]" should be updated from "\"Java version: 11.\"" to "\"Java version: 17.\"", in file "cst.yml"
```

**Pull request merged**